node/network/availability-distribution/src/lib.rs

```diff
@@ -365,13 +365,7 @@ where
 			.map(|(peer, _view)| peer.clone())
 			.collect();
 
-		// If we have no peers that are interested, skip the rest.
-		if peers.is_empty() {
```

Comment:
It's not clear what value we get from not `continue`ing here.

Reply:
This was the lengthy discussion about adding the message to the `message_vault`, which is a goto cache for seen and messages to redistribute. Not having it in the store, because receiving it at a time where none of the peers has interest in the particular candidate can lead to duplicate gossips which will harm reputation.